### PR TITLE
Some misc fixes on CSI driver

### PR DESCRIPTION
**Pull request by ngphibang**
This PR adds some misc fixes for CSI driver. It is splitted from https://github.com/zephyrproject-rtos/zephyr/pull/69810 to ease the review process as required by @loicpoulain.

It already got reviewed by Daniel Degrasse.

### CODE REVIEW

**Review by decsny — DISMISSED** [review]

File: drivers/video/video_mcux_csi.c [hunk 1]
```diff
@@ -126,7 +126,6 @@ static int video_mcux_csi_set_fmt(const struct device *dev, enum video_endpoint_
 		return -EINVAL;
 	}
 
-	data->pixelformat = fmt->pixelformat;
```
Comment (danieldegrasse) [inline drivers/video/video_mcux_csi.c:132]: Can we drop the `pixelformat` field from the `data` structure as well?

**Comment by dleach02** [comment]
@ngphibang please address the CI problem

**Comment by ngphibang** [comment]
@dleach02 The CI problem is not due to my changes. In fact, it is because the "Install Packages" step failed.

File: drivers/video/video_mcux_csi.c [hunk 2]
```diff
@@ -166,14 +165,9 @@ static int video_mcux_csi_get_fmt(const struct device *dev, enum video_endpoint_
 	if (config->sensor_dev && !video_get_format(config->sensor_dev, ep, fmt)) {
 		/* align CSI with sensor fmt */
 		return video_mcux_csi_set_fmt(dev, ep, fmt);
+	} else {
```
Comment (loicpoulain) [inline drivers/video/video_mcux_csi.c:167]: ok, but no need for a 'else' here.

File: drivers/video/video_mcux_csi.c (hunk 1 above)
Comment (ngphibang) [reply drivers/video/video_mcux_csi.c:132]: yes, I forgot this, thanks

File: drivers/video/video_mcux_csi.c (hunk 2 above)
Comment (ngphibang) [reply drivers/video/video_mcux_csi.c]: yes, thanks

**Comment by ngphibang** [comment]
@danieldegrasse @loicpoulain : Thanks for the comments. Is it ok for you now ?

**Review by danieldegrasse — APPROVED** [review]

**Review by decsny — APPROVED** [review]

**Review by dleach02 — APPROVED** [review]

**Pull request merged**